packages/fxa-shared/subscriptions/metadata.ts

```diff
@@ -35,20 +35,30 @@ export const DEFAULT_PRODUCT_DETAILS: ProductDetails = {
 
 // Support some default null values for product / plan metadata and
 // allow plan metadata to override product metadata
-export const metadataFromPlan = (plan: Plan): ProductMetadata => ({
-  productSet: null,
-  productOrder: null,
-  emailIconURL: null,
-  webIconURL: null,
-  webIconBackground: null,
-  upgradeCTA: null,
-  successActionButtonURL: null,
-  'product:termsOfServiceDownloadURL': '',
-  'product:termsOfServiceURL': '',
-  'product:privacyNoticeURL': '',
-  ...plan.product_metadata,
-  ...plan.plan_metadata,
-});
+export const metadataFromPlan = (plan: Plan): ProductMetadata => {
+  const metadata = {
+    productSet: null,
+    productOrder: null,
+    emailIconURL: null,
+    webIconURL: null,
+    webIconBackground: null,
+    upgradeCTA: null,
+    successActionButtonURL: null,
+    'product:termsOfServiceDownloadURL': '',
+    'product:termsOfServiceURL': '',
+    'product:privacyNoticeURL': '',
+    ...plan.product_metadata,
+    ...plan.plan_metadata,
+  };
+
+  if (typeof metadata.productSet === 'string') {
+    //@ts-ignore
+    // prettier-ignore
+    metadata.productSet = metadata.productSet.trim().split(',').map((elem: string) => elem.trim()).filter((elem) => elem !== '');
```

Comment:
We should probably have a `commaSeparatedStringToArray` method at some point. 😉

Reply:
Yeah there is a `commaSeparatedListToArray` method in `fxa-auth-server` but I didn't want to create a circular reference between auth and shared so I moved it in a new utils file in shared.